_rules/css-orientation-lock-media-queries-b33eff.md

```diff
@@ -0,0 +1,209 @@
+---
+id: b33eff
+name: Page has no orientation lock specified using CSS media queries
+rule_type: atomic
+description: |
+  This rule checks that page content is not locked to any single display orientation using CSS media queries.
+accessibility_requirements:
+ wcag20:1.3.4: # Orientation
+  forConformance: true
+  failed: not satisfied
+  passed: further testing needed
+  inapplicable: further testing needed
+input_aspects:
+- DOM Tree
+- CSS Styling
+authors:
+- Jey Nandakumar
+- Audrey Maniez
+---
+
+## Applicability
+
+The rule applies to any element that is [visible](#visible) and has a CSS [transform](https://developer.mozilla.org/en-US/docs/Web/CSS/transform) property with any of the below [transform-functions](https://developer.mozilla.org/en-US/docs/Web/CSS/transform-function)
+
+- [rotate transform-function](https://developer.mozilla.org/en-US/docs/Web/CSS/transform-function/rotate)
+- [rotate3d transform-function](https://developer.mozilla.org/en-US/docs/Web/CSS/transform-function/rotate3d)
+- [rotateX transform-function](https://developer.mozilla.org/en-US/docs/Web/CSS/transform-function/rotateX)
+- [rotateY transform-function](https://developer.mozilla.org/en-US/docs/Web/CSS/transform-function/rotateY)
+- [rotateZ transform-function](https://developer.mozilla.org/en-US/docs/Web/CSS/transform-function/rotateZ)
+- [matrix transform-function](https://developer.mozilla.org/en-US/docs/Web/CSS/transform-function/matrix)
+- [matrix3d transform-function](https://developer.mozilla.org/en-US/docs/Web/CSS/transform-function/matrix3d)
+
+which are set as CSS [media query](https://developer.mozilla.org/en-US/docs/Web/CSS/Media_Queries) with an [orientation](https://developer.mozilla.org/en-US/docs/Web/CSS/@media/orientation) [media feature](https://developer.mozilla.org/en-US/docs/Web/CSS/Media_Queries/Using_media_queries#Media_features) with a value of `landsapce` or `portrait`.
```

Comment:
> [...] which are set as CSS [media query](https://developer.mozilla.org/en-US/docs/Web/CSS/Media_Queries) [...]

This should be clarified as there are several uses of media queries in CSS. One is through `@media` rules as exemplified in the test cases, but `@import` rules can also be conditional:

```css
@import "./lock-orientation.css" (orientation: portrait);
```

The `<link>` element can also be made conditional:

```html
<link rel="stylesheet" media="(orientation: portrait)" href="./lock-orientation.css">
```

Perhaps something along the lines of:

> that are applied conditionally based on the [orientation](https://www.w3.org/TR/mediaqueries/#orientation) media feature.

No yet entirely unambiguous, but I think it's a start.

Reply:
This is true. Not emphasising just via media queries is important.